### PR TITLE
Dev architecture

### LINKED ISSUE #23: Learn about and build a solid coding architecture for the app

### PR DESCRIPTION
**Pull request by WollfieGitHub**
Closes #23

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -21,9 +21,11 @@ kotlin.code.style=official
...
 android.nonTransitiveRClass=true
-
 # Implement CI (done by ourselves)
 org.gradle.daemon=true
 org.gradle.caching=true
 org.gradle.parallel=true
 org.gradle.configureondemand=true
+API_KEY="API_KEY"
+BASE_URL="BASE_URL"
+POSTER_URL="POSTER_URL"
```
Comment (Lynoxe) [inline gradle.properties:31]: We don't need API_KEY and POSTER_URL, we will probably delete them

**Review by Lynoxe — COMMENTED** [review]
We just need to make sure that the CodeClimate coverage is correctly setup

File: .cirrus.yml [hunk 2]
```diff
@@ -40,9 +40,20 @@ check_android_task:
   check_script:
     ./gradlew check connectedCheck
   report_codeclimate_script: |
+    ls ./app/build/reports/jacoco/jacocoTestReport
+    ls ./data/build/
+    ls ./data/build/reports/jacoco/jacocoTestReport
+    ls ./domain/build/
```
[the remark this answers is not in the document]
Comment (WollfieGitHub) [reply .cirrus.yml:46]: This was just to make sure the necessary files were created by jacoco but I thought I deleted it

**Comment by WollfieGitHub** [comment]
image
I think Code Climate really seems to be working given the "100%" coverage on this file from the data submodule (Which indeed has one unit test)

File: gradle.properties (hunk 1 above)
Comment (WollfieGitHub) [reply gradle.properties]: Changed in next commit

**Review by Lynoxe — APPROVED** [review]
I confirm that CodeClimate use all the files for its coverage, so it's all good for me

**Pull request merged**